kernel/inferCumulativity.ml

```diff
@@ -15,30 +15,74 @@ open Univ
 open Variance
 open Util
 
-type inferred = IrrelevantI | CovariantI
-
-(** Throughout this module we modify a map [variances] from local
-    universes to [inferred]. It starts as a trivial mapping to
-    [Irrelevant] and every time we encounter a local universe we
-    restrict it accordingly.
-    [Invariant] universes are removed from the map.
-*)
 exception TrivialVariance
+exception BadVariance of Level.t
```

Comment:
I think we should enrich BadVariance with the expected and infered variance to provide a more meaningful error message.

Reply:
Updated